x-pack/platform/plugins/shared/security/server/session_management/session_index.ts

```diff
@@ -625,16 +625,37 @@ export class SessionIndex {
     );
   }
 
+  private async attachAliasToIndex() {
+    // Prior to https://github.com/elastic/kibana/pull/134900, sessions would be written directly against the session index.
+    // Now, we write sessions against a new session index alias. This call ensures that the alias exists, and is attached to the index.
+    // This operation is safe to repeat, even if the alias already exists. This seems safer than retrieving the index details, and inspecting
+    // it to see if the alias already exists.
+    try {
+      await this.options.elasticsearchClient.indices.putAlias({
+        index: this.indexName,
+        name: this.aliasName,
+      });
+    } catch (err) {
+      this.options.logger.error(`Failed to attach alias to session index: ${err.message}`);
+      throw err;
+    }
+  }
+
   /**
    * Creates the session index if it doesn't already exist.
    */
   private async ensureSessionIndexExists() {
     // Check if required index exists.
     let indexExists = false;
     try {
-      indexExists = await this.options.elasticsearchClient.indices.exists({
+      const aliasNameExists = await this.options.elasticsearchClient.indices.exists({
         index: this.aliasName,
       });
+      const indexNameExists = await this.options.elasticsearchClient.indices.exists({
+        index: this.indexName,
+      });
+
+      indexExists = aliasNameExists || indexNameExists;
```

Comment:
__question:__ Would you mind clarifying in the code comment which case we’re handling here and why?

__question:__ Do we need this change in `main` or it'd be sufficient to have this change only in `8.x` considering that it's not possible to use 7.x data with 9.0 directly without going through UA? Is there any benefits in having this change in `main` that I'm missing?